website_sharing_whole_class_mosaic_plot_translation.Rmd

```diff
@@ -0,0 +1,507 @@
+
+* conds : 选择一个变量作为条件
+
+写为conds = product(cond1, cond2, ...)
+
+然后通过productplots函数发送这些值，以创建所需分布的公式。
+
+```{r}
+#公式：
+weight ~ fill + x | conds
+```
+
+
+#### 3.41 从规则到代码#
+
+如何写成代码
+
+* weight=1
+
+* x = product(Y, X)
+
+* fill=W
+
+* conds = product(Z)
+
+* 这些美学设置了分配的公式。
+
+#### 3.42 1 ~ X
+
+```{r}
+library(ggplot2)
+library(ggmosaic)
+library(gridExtra)
+ ggplot(data = fly) +
+   geom_mosaic(aes(x = product(RudeToRecline), fill=RudeToRecline), na.rm=TRUE) +
+   labs(x="Is it rude recline? ", title='f(RudeToRecline)') 
+```
+
+#### 3.43 1 ~ Y + X
+
+```{r}
+ggplot(data = fly) +
+   geom_mosaic(aes(x = product(DoYouRecline, RudeToRecline), fill=DoYouRecline), na.rm=TRUE) + 
+  labs(x = "Is it rude recline? ", title='f(DoYouRecline | RudeToRecline) f(RudeToRecline)')
+```
+
+#### 3.44 1 ~ X + Y / Z
+
+```{r}
+ggplot(data = fly) +
+   geom_mosaic(aes(x = product(DoYouRecline, RudeToRecline), fill=DoYouRecline, conds=product(Gender)), na.rm=TRUE, divider=mosaic("v")) +  labs(x = "Is it rude recline? ", title='f(DoYouRecline, RudeToRecline| Gender)')
+```
+
+#### 3.45 Alternative to conditioning: facetting
+
+```{r}
+ggplot(data = fly) +
+   geom_mosaic(aes(x = product(DoYouRecline, RudeToRecline), fill=DoYouRecline), na.rm=TRUE) +  labs(x = "Is it rude recline? ", title='f(DoYouRecline, RudeToRecline| Gender)') + facet_grid(Gender~.)
+```
+
+
+### 3.5 Importance of ordering
+
+```{r}
+ ggplot(data = fly) +
+   geom_mosaic(aes(x = product(DoYouRecline, RudeToRecline), fill=DoYouRecline), na.rm=TRUE) +  labs(x = "Is it rude recline? ", title='f(DoYouRecline | RudeToRecline) f(RudeToRecline)') + theme(plot.title = element_text(size = rel(1)))
+
+ ggplot(data = fly) +
+   geom_mosaic(aes(x = product(RudeToRecline, DoYouRecline), fill=DoYouRecline), na.rm=TRUE) + labs(x = "" , y = "Is it rude recline? ", title='f(DoYouRecline | RudeToRecline) f(RudeToRecline)') + coord_flip() + theme(plot.title = element_text(size = rel(1)))
+
+par(mfrow=c(2,1))
+```
+
+
+
+
+### 3.6 geom_mosaic的其他特征
+
+geom_mosaic独有的参数。
+
+* divider: 用于声明要使用的分区的类型。
+* offset：设置第一条spine之间的空间。
+
+### 3.7 Divider function。分割类型。
+每个部分有四个选项。
+
+* vspine：宽度不变，高度可变。
+* hspine：高度不变，宽度可变。
+* vbar：高度不变，宽度不同。
+* hbar：宽度不变，高度不同。
+
+```{r}
+hbar <- ggplot(data = fly) +
+   geom_mosaic(aes(x = product(FlightFreq), fill=FlightFreq), divider="hbar", na.rm=TRUE) + labs(x=" ", title='divider = "hbar"') 
+
+
+hspine <- ggplot(data = fly) +
+   geom_mosaic(aes(x = product(FlightFreq), fill=FlightFreq),  divider="hspine", na.rm=TRUE) + labs(x=" ", title='divider = "hspine"') 
+
+
+vbar <- ggplot(data = fly) +
+   geom_mosaic(aes(x = product(FlightFreq), fill=FlightFreq), divider="vbar", na.rm=TRUE) + labs(y=" ", x="", title='divider = "vbar"') 
+ 
+vspine <- ggplot(data = fly) +
+   geom_mosaic(aes(x = product(FlightFreq), fill=FlightFreq), divider="vspine", na.rm=TRUE) + labs(y=" ", x="", title='divider = "vspine"') 
+
+hbar
+hspine 
+vbar
+vspine
+```
+
+### 3.8 使用一个或多个变量进行分区
```

Comment:
Hashtag headers should not be followed by numbers since the hashtags will create numbered subheadings.